### PR TITLE
New shape samples

### PR DESCRIPTION
**Pull request by gregwoo-microsoft**
New samples to show how to create various shapes including:
- Ellipses
- Rectangles
- Polygons
- Lines
- Polylines
- Paths
- Geometry groups

Note: This work is in progress

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by gregwoo-microsoft** [comment]
@chingucoding let me know what you think. Note: this is still a work in progress but I'd love to get your early feedback. For instance, I think it would make more sense to break up those samples into multiple pages (e.g. Shapes, Lines, Paths). Also, I couldn't figure out how to bind a Polygon's point X/Y coordinate to a slider value. Let me know if you know how to do that, otherwise, I'll have another look at it next week.

**Comment by chingucoding** [comment]
I think separating them out would be a good idea, maybe they should get their own "Shapes" section where we give Rectangle, Line, ... all their own page?

Regarding polygon point X/Y, the only way to change them it seems is through point animations (https://learn.microsoft.com/uwp/api/windows.foundation.point?view=winrt-22621&WT.mc_id=WDIT-MVP-5004239#animating-point-values) but even them I'm not entirely sure it will work. If I remember correctly, some of these APIs take a snapshot of the points and after that, when the entries such as points get modified they don't take effect. But maybe I would still try out the point animations if we want to change the points position.

**Comment by gregwoo-microsoft** [comment]
@chingucoding I separated the samples into 2 pages: 1st page for shapes (rectangle, ellipse, polygon) and a 2nd page for lines (line, polyline, path, geometry group). I put those 2 pages into the "Styles" subsection. Let me know what you think.

File: WinUIGallery/ControlPages/ShapePage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        //    var newWindow = WindowHelper.CreateWindow();
+        //    var rootPage = new NavigationRootPage();
+        //    rootPage.RequestedTheme = ThemeHelper.RootTheme;
+        //    newWindow.Content = rootPage;
+        //    newWindow.Activate();
+
+        //    var targetPageType = typeof(NewControlsPage);
+        //    string targetPageArguments = string.Empty;
+        //    rootPage.Navigate(targetPageType, targetPageArguments);
+        //}
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/ShapePage.xaml.cs:43]: This can be removed.

File: WinUIGallery/ControlPages/LinePage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        //    var newWindow = WindowHelper.CreateWindow();
+        //    var rootPage = new NavigationRootPage();
+        //    rootPage.RequestedTheme = ThemeHelper.RootTheme;
+        //    newWindow.Content = rootPage;
+        //    newWindow.Activate();
+
+        //    var targetPageType = typeof(NewControlsPage);
+        //    string targetPageArguments = string.Empty;
+        //    rootPage.Navigate(targetPageType, targetPageArguments);
+        //}
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/LinePage.xaml.cs:43]: This can be deleted.

File: WinUIGallery/ControlPages/LinePage.xaml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            <local:ControlExample.Options>
+                <StackPanel Width="220">
+                    <Slider x:Name="lineSlider1" Header="Point #1 X" Minimum="0" Maximum="100" 
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                    <Slider x:Name="lineSlider2" Header="Point #1 Y" Minimum="0" Maximum="100" 
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                    <Slider x:Name="lineSlider3" Header="Point #2 X" Minimum="200" Maximum="300" 
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                    <Slider x:Name="lineSlider4" Header="Point #2 Y" Minimum="0" Maximum="100" 
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/LinePage.xaml:41]: How about naming those "Startpoint X, Startpoint Y,..."?

File: WinUIGallery/ControlPages/LinePage.xaml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                    <ToggleSwitch x:Name="ToggleSwitch2" Header="Show points" IsOn="False" />
+                    <Slider x:Name="polyLineSlider1" Header="Stroke Thickness" Minimum="2" Maximum="10"
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                </StackPanel>
+            </local:ControlExample.Options>
+
+            <local:ControlExample.Xaml>
+                <x:String xml:space="preserve">
+&lt;Polyline Stroke="Black" StrokeThickness="$(Slider1)"
+Points="10,100 60,40 200,40 250,100"/&gt;
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/LinePage.xaml:97]: nit: Formatting

File: WinUIGallery/ControlPages/LinePage.xaml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                </StackPanel>
+            </local:ControlExample.Options>
+
+            <local:ControlExample.Xaml>
+                <x:String xml:space="preserve">
+&lt;Path Stroke="DarkGoldenRod" StrokeThickness="$(Slider1)"
+Data="M 10,100 C 100,25 300,250 400,75 H 200"/&gt;
+
+&lt;!-- The first segment is a cubic Bezier curve that begins at Point #1 and ends at Point #4, which is drawn by using Point #2 and 3 as the two control points. This segment is indicated by the "C" command in the Data attribute string. --&gt;
+&lt;!-- The second segment begins with an absolute horizontal line command "H", which specifies a line drawn from the preceding subpath endpoint (Point #4) to a new endpoint (Point #5). Because it's a horizontal line command, the value specified is an x-coordinate. --&gt;
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/LinePage.xaml:141]: How about putting this above the control? I think most people would expect comments to be before the code they are documenting.

File: WinUIGallery/ControlPages/LinePage.xaml [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                    <ToggleSwitch x:Name="ToggleSwitch" Header="Show points" IsOn="False" />
+                    <Slider x:Name="pathSlider1" Header="Stroke Thickness" Minimum="2" Maximum="10"
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                </StackPanel>
+            </local:ControlExample.Options>
+
+            <local:ControlExample.Xaml>
+                <x:String xml:space="preserve">
+&lt;Path Stroke="DarkGoldenRod" StrokeThickness="$(Slider1)"
+Data="M 10,100 C 100,25 300,250 400,75 H 200"/&gt;
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/LinePage.xaml:138]: nit: formatting

File: WinUIGallery/ControlPages/ShapePage.xaml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+                    <Slider x:Name="recSlider5" Header="Radius X" Minimum="0" Maximum="100"
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                </StackPanel>
+            </local:ControlExample.Options>
+
+            <local:ControlExample.Xaml>
+                <x:String xml:space="preserve">
+&lt;Rectangle Fill="SteelBlue" Height="$(Slider1)" Width="$(Slider2)"
+Stroke="Black" StrokeThickness="$(Slider3)"
+RadiusY="$(Slider4)" RadiusX="$(Slider5)"/&gt;
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/ShapePage.xaml:89]: nit: formatting

File: WinUIGallery/ControlPages/ShapePage.xaml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                    <ToggleSwitch x:Name="ToggleSwitchPoly" Header="Show points" IsOn="False" />
+                    <Slider x:Name="polySlider1" Header="Stroke Thickness" Minimum="2" Maximum="10"
+                        StepFrequency="0.5" SmallChange="1" IsFocusEngagementEnabled="False"/>
+                </StackPanel>
+            </local:ControlExample.Options>
+
+            <local:ControlExample.Xaml>
+                <x:String xml:space="preserve">
+&lt;Polygon Fill="SteelBlue" Points="10,100 60,40 200,40 250,100"
+StrokeThickness="$(Slider1)" Stroke="Black"/&gt;
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/ShapePage.xaml:135]: nit: formatting

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 9]
```diff
@@ -2226,47 +2270,53 @@
...
-          "Content": "",
+          "UniqueId": "Line",
+          "Title": "Line",
+          "ApiNamespace": "Microsoft.UI.Xaml.Shapes",
+          "Subtitle": "Draws a straight line between two points.",
+          "ImagePath": "ms-appx:///Assets/ControlImages/CreateMultipleWindows.png",
+          "ImageIconPath": "ms-appx:///Assets/ControlIcons/DefaultIcon.png",
+          "Description": "Draws a straight line between two points.",
+          "Content": "<p>Look at the <i>LinePage.xaml</i> file in Visual Studio to see the full code for this page.</p>",
           "IsNew": false,
```
Comment (chingucoding) [inline WinUIGallery/DataModel/ControlInfoData.json:2306]: How about marking this as new?

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 10]
```diff
@@ -2226,47 +2270,53 @@
...
+          "Title": "Shape",
+          "ApiNamespace": "Microsoft.UI.Xaml.Shapes",
+          "Subtitle": "How to draw shapes, such as ellipses, rectangles, and polygons.",
+          "ImagePath": "ms-appx:///Assets/ControlImages/CreateMultipleWindows.png",
           "ImageIconPath": "ms-appx:///Assets/ControlIcons/DefaultIcon.png",
-          "Description": "Apply custom colors to your controls through this cascading API, or scope them to a local subset.",
-          "Content": "",
+          "Description": "Basic shapes are intended for decorative rendering or for compositing non-interactive parts of controls.",
+          "Content": "<p>Look at the <i>ShapePage.xaml</i> file in Visual Studio to see the full code for this page.</p>",
           "IsNew": false,
```
Comment (chingucoding) [inline WinUIGallery/DataModel/ControlInfoData.json:2281]: How about marking this as new?

**Comment by chingucoding** [comment]
Left a few comments, mostly small things. I really like the new samples, especially the idea of having a toggle to show the points the polyline/polygon is made of.

File: WinUIGallery/ControlPages/ShapePage.xaml (hunk 7 above)
Comment (gregwoo-microsoft) [reply WinUIGallery/ControlPages/ShapePage.xaml:89]: @chingucoding is it just a question of adding a tab for the 2nd and 3rd lines or is there something else?

File: WinUIGallery/ControlPages/LinePage.xaml (hunk 5 above)
Comment (gregwoo-microsoft) [reply WinUIGallery/ControlPages/LinePage.xaml:138]: Done!

**Comment by gregwoo-microsoft** [comment]
@chingucoding I implemented the changes you suggested. Let me know if you think that there is something else to add/modify 😄

**Review by chingucoding — APPROVED** [review]
LGTM, really great that we also showcase these APIs now!

**Comment by gregwoo-microsoft** [comment]
The only things missing are the 2 control icons, I'll work with our design team to get those.

**Pull request merged**